OrbitCore/LinuxUtils.cpp

```diff
@@ -86,39 +88,52 @@ std::string ExecuteCommand(const char* a_Cmd) {
 //-----------------------------------------------------------------------------
 void ListModules(pid_t a_PID,
                  std::map<uint64_t, std::shared_ptr<Module>>& o_ModuleMap) {
-  std::map<std::string, std::shared_ptr<Module>> modules;
+  std::map<std::string, std::pair<uint64_t, uint64_t>> module_addresses;
   std::vector<std::string> result = ListModules(a_PID);
+
   for (const std::string& line : result) {
     std::vector<std::string> tokens = Tokenize(line);
-    if (tokens.size() == 6) {
-      const std::string& moduleName = tokens[5];
-
-      auto addresses = Tokenize(tokens[0], "-");
-      if (addresses.size() == 2) {
-        auto iter = modules.find(moduleName);
-        std::shared_ptr<Module> module =
-            (iter == modules.end()) ? std::make_shared<Module>() : iter->second;
-
-        uint64_t start = std::stoull(addresses[0], nullptr, 16);
-        uint64_t end = std::stoull(addresses[1], nullptr, 16);
-
-        if (module->m_AddressStart == 0 || start < module->m_AddressStart)
-          module->m_AddressStart = start;
-        if (end > module->m_AddressEnd) module->m_AddressEnd = end;
-
-        module->m_FullName = moduleName;
-        module->m_Name = Path::GetFileName(module->m_FullName);
-        module->m_Directory = Path::GetDirectory(module->m_FullName);
-        module->m_PdbSize = Path::FileSize(moduleName);
-        modules[moduleName] = module;
-      }
+
+    // tokens[4] is the inode column. If inode equals 0, then the memory is not
+    // mapped to a file (might be heap, stack or something else)
+    if (tokens.size() != 6 || tokens[4] == "0") continue;
+
+    auto addresses = Tokenize(tokens[0], "-");
+    if (addresses.size() != 2) continue;
+
+    const std::string& module_name = tokens[5];
+
+    uint64_t start = std::stoull(addresses[0], nullptr, 16);
+    uint64_t end = std::stoull(addresses[1], nullptr, 16);
+
+    auto iter = module_addresses.find(module_name);
+    if (iter == module_addresses.end()) {
+      module_addresses[module_name] = std::pair(start, end);
     }
+    std::pair<uint64_t, uint64_t>& module_address =
+        module_addresses[module_name];
+    if (start < module_address.first) module_address.first = start;
+    if (end > module_address.second) module_address.second = end;
```

Comment:
This should probably go to else, you can also use iter here instead of [], in which case also reverse condition - check != end first

Reply:
done